src/Psl/Result/Success.php

```diff
@@ -44,6 +44,20 @@ public function getResult(): mixed
         return $this->value;
     }
 
+    /**
+     * Unwrap the Result if it is succeeded or return $default value.
+     *
+     * @param D $default
+     *
+     * @return T|D
+     *
+     * @template D
+     */
+    public function unwrapOr(mixed $default): mixed
+    {
+        return $this->value;
+    }
```

Comment:
this always returns `T`.

```suggestion
     * @template D
     *
     * @param D $default
     *
     * @return T
     */
    public function unwrapOr(mixed $default): mixed
    {
        return $this->value;
    }
```